src/initialization/MOM_state_initialization.F90

```diff
@@ -234,8 +239,12 @@ subroutine MOM_initialize_state(u, v, h, tv, Time, G, GV, US, PF, dirs, &
     if (.NOT.use_temperature) call MOM_error(FATAL,"MOM_initialize_state : "//&
        "use_temperature must be true if INIT_LAYERS_FROM_Z_FILE is true")
 
-    call MOM_temp_salt_initialize_from_Z(h, tv, G, GV, US, PF, just_read_params=just_read)
-
+    if (use_ice_shelf) then
```

Comment:
Because frac_shelf_h is an optional argument that is just being passed through, these 5 lines could be replaced by just the call on lines 243-244.